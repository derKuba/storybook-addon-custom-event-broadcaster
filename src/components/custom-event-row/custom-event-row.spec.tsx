import React from 'react';
import renderer from 'react-test-renderer';

import CustomEventRow from './custom-event-row';

jest.mock('@storybook/components', () => ({
  __esModule: true,
  Icons: ({ children }) => <div>{children}</div>,
  Button: ({ onClick, children }) => {
    console.log(onClick, children);
    return <button onClick={onClick}>{children}</button>;
  },
}));

const dispatchMock = jest.fn();
global.document.getElementById = () => ({
  contentWindow: {
    document: {
      querySelector: () => ({
        dispatchEvent: dispatchMock,
      }),
      dispatchEvent: dispatchMock,
    },
  },
});

describe('CustomEventRow', () => {
  test('should render', () => {
    const tree = renderer
      .create(
        <CustomEventRow
          selectorDefault=".test"
          eventNameDefault="test:event"
          eventDataDefault={null}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should click the button and send event with selector', () => {
    const tree = renderer.create(
      <CustomEventRow
        selectorDefault=".test"
        eventNameDefault="test:event"
        eventDataDefault={null}
      />,
    );

    tree.root.findByType('button').props.onClick();

    expect(dispatchMock).toBeCalled();
  });

  test('should click the button and send event without selector', () => {
    const tree = renderer.create(
      <CustomEventRow eventNameDefault="test:event" eventDataDefault={null} />,
    );

    tree.root.findByType('button').props.onClick();

    expect(dispatchMock).toBeCalled();
  });
});
