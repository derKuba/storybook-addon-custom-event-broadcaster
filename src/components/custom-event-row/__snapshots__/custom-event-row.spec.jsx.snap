// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomEventRow should render 1`] = `
<tr>
  <td>
    <input
      className="css-12d6kl5"
      onChange={[Function]}
      type="text"
      value="test:event"
    />
  </td>
  <td>
    <textarea
      className="css-12d6kl5"
      onChange={[Function]}
      value={null}
    />
  </td>
  <td>
    <div
      className="css-k008qs"
    >
      <textarea
        className="css-12d6kl5"
        onChange={[Function]}
        value=".test"
      />
    </div>
  </td>
  <td>
    <button
      onClick={[Function]}
    >
      send
    </button>
  </td>
</tr>
`;
